core/src/main/scala/org/apache/spark/deploy/history/HybridStore.scala

```diff
@@ -144,10 +146,9 @@ private[history] class HybridStore extends KVStore {
     backgroundThread = new Thread(() => {
       try {
         for (klass <- klassMap.keys().asScala) {
-          val it = inMemoryStore.view(klass).closeableIterator()
-          while (it.hasNext()) {
-            levelDB.write(it.next())
-          }
+          val values = Lists.newArrayList(
```

Comment:
This would be OK, given all entries are from inMemoryStore which are already materialized into memory.